src/core/src/runtime/ov_tensor.cpp

```diff
@@ -98,7 +99,14 @@ element::Type Tensor::get_element_type() const {
 }
 
 void Tensor::set_shape(const ov::Shape& shape) {
-    OV_TENSOR_STATEMENT(_impl->setShape({shape.begin(), shape.end()}));
+    // WA for tensor conversion from host tensor with dynamic shape.
```

Comment:
is it a temporary WA until remove legacy code?

Reply:
Yes it is. It is needed in order to support set_shape for tensors which were created from ngraph dynamic tensors